File: app/src/main/java/app/cloudmusic/utils/MediaSharePreference.java
package app.cloudmusic.utils;

/**
 * Created by Administrator on 2018/3/31.
 */

public class MediaSharePreference extends BaseSPF {
    private static MediaSharePreference sharePreference = null;

    public MediaSharePreference() {
        super("AppBasicData");
    }

    public static MediaSharePreference getInstances(){
        if (sharePreference == null)
            sharePreference = new MediaSharePreference();
        return sharePreference;
    }

    public void setRepeatMode(int repeatMode){
        writeInt("repeatMode",repeatMode);
    }

    public int getRepeatMode(){
        return readInt("repeatMode",1);
    }
}

File: app/src/main/java/app/cloudmusic/utils/MediaSharePreference.kt
package app.cloudmusic.utils

/**
 * Created by Administrator on 2018/3/31.
 */

class MediaSharePreference : BaseSPF("AppBasicData") {

    var repeatMode: Int
        get() = readInt("repeatMode", 1)
        set(repeatMode) = writeInt("repeatMode", repeatMode)

    companion object {
        private var sharePreference: MediaSharePreference? = null

        val instances: MediaSharePreference
            get() {
                if (sharePreference == null)
                    sharePreference = MediaSharePreference()
                return sharePreference!!
            }
    }
}
